src/protocols/bdx/BDXMessageUtils.cpp

```diff
@@ -185,22 +158,35 @@ CHIP_ERROR TransferInit::Parse(System::PacketBuffer * aBuffer, TransferInit & aP
 
     SuccessOrExit(bufReader.Read16(&aParsedMessage.mFileDesLength).StatusCode());
 
-    VerifyOrExit(bufReader.OctetsRead() + aParsedMessage.mFileDesLength <= aBuffer->DataLength(),
+    VerifyOrExit(bufReader.OctetsRead() + aParsedMessage.mFileDesLength <= aBuffer.DataLength(),
```

Comment:
Suggest `(aBuffer.DataLength() - bufReader.OctetsRead()) >= aParsedMessage.mFileDesLength` to be resilient to wraparound in a cleaner way. In fact, there ought to be a `bufReader.Remaining()` call that is gives you `aBuffer.DataLength() - bufReader.OctetsRead()`. You should *never* have to do your own subtraction for buffer space remaining, as it is very error-prone.

The code should read `VerifyOrExit(buffer.Remaining() >= aParsedMessage.mFileDesLength)`

@bzbarsky-apple Suggest adding `Remaining()` in `Reader` to avoid these computations.

Reply:
Yep, I had ended up adding an `Available()` in `Reader` yesterday for something else I was noodling on because some code wanted to print the size of the incoming data.  That particular change is not landing any time soon for other reasons, but it absolutely makes sense to me to add some sort of "how many bytes do we have left?" API on `Reader`.